cgo/cgo.go

```diff
@@ -1142,27 +1152,55 @@ func (f *cgoFile) getASTDeclNode(name string, found clangCursor, iscall bool) as
 			// Original cgo reports an error like
 			//   cgo: inconsistent definitions for C.myint
 			// which is far less helpful.
-			f.addError(getPos(node), "defined previously at "+f.fset.Position(getPos(newNode)).String()+" with a different type")
+			f.addError(getPos(node), name+" defined previously at "+f.fset.Position(getPos(newNode)).String()+" with a different type")
 		}
 		f.defined[name] = node
 		return node
 	}
 
 	// The declaration has no AST node. Create it now.
 	f.defined[name] = nil
-	node, elaboratedType := f.createASTNode(name, found)
+	node, extra := f.createASTNode(name, found)
 	f.defined[name] = node
-	f.definedGlobally[name] = node
 	switch node := node.(type) {
 	case *ast.FuncDecl:
+		if strings.HasPrefix(node.Doc.List[0].Text, "//export _Cgo_static_") {
+			// Static function. Only accessible in the current Go file.
+			globalName := strings.TrimPrefix(node.Doc.List[0].Text, "//export ")
+			// Make an alias. Normally this is done using the alias function
+			// attribute, but MacOS for some reason doesn't support this (even
+			// though the linker has support for aliases in the form of N_INDR).
+			// Therefore, create an actual function for MacOS.
+			var params []string
+			for _, param := range node.Type.Params.List {
+				params = append(params, param.Names[0].Name)
+			}
+			callInst := fmt.Sprintf("%s(%s);", name, strings.Join(params, ", "))
+			if node.Type.Results != nil {
+				callInst = "return " + callInst
+			}
+			aliasDeclaration := fmt.Sprintf(`
+#ifdef __APPLE__
```

Comment:
:crying_cat_face: oh Apple

Reply:
The worst part is that the linker actually _does_ support aliases, it's just the compiler that doesn't.
...and I don't feel like creating a Clang patch and pushing for its inclusion.